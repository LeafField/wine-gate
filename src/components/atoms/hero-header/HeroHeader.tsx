import React, { FC } from "react";

type Props = {
  title: string;
};

const HeroHeader: FC<Props> = ({ title }) => {
  return (
    <div className="absolute inset-0 z-front flex h-full w-full items-center justify-end hero-mobile-overlay tablet:hero-overlay">
      <div className="mx-auto flex w-full max-w-[87rem] items-center justify-end px-6">
        <h2 className="text-heading2 font-bold text-white">{title}</h2>
      </div>
    </div>
  );
};

export default HeroHeader;
