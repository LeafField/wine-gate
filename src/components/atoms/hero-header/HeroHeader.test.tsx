import { screen, render } from "@testing-library/react";
import HeroHeader from "./HeroHeader";

describe("HeroHeaderの単体テスト", () => {
  test("propsのtitleが表示される", () => {
    const { container } = render(<HeroHeader title="新着ワイン" />);
    expect(
      screen.getByRole("heading", { name: "新着ワイン" }),
    ).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
