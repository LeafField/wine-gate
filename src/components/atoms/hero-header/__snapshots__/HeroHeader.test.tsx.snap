// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroHeaderの単体テスト propsのtitleが表示される 1`] = `
<div>
  <div
    class="absolute inset-0 z-front flex h-full w-full items-center justify-end hero-mobile-overlay tablet:hero-overlay"
  >
    <div
      class="mx-auto flex w-full max-w-[87rem] items-center justify-end px-6"
    >
      <h2
        class="text-heading2 font-bold text-white"
      >
        新着ワイン
      </h2>
    </div>
  </div>
</div>
`;
